tests/rngtest/main.c

```diff
@@ -0,0 +1,347 @@
+        fips_test_store(ctx, rngdatabuf[i + 3]);
+    }
+
+    /* Add in the last (possibly incomplete) run */
+    if (ctx->rlength < 5) {
+        ctx->runs[ctx->rlength + (6 * ctx->current_bit)]++;
+    }
+    else {
+        ctx->runs[5 + (6 * ctx->current_bit)]++;
+
+        if (ctx->rlength >= 25) {
+            rng_test |= FIPS_RNG_LONGRUN;
+        }
+    }
+
+    if (ctx->longrun) {
+        rng_test |= FIPS_RNG_LONGRUN;
+        ctx->longrun = 0;
+    }
+
+    /* Ones test */
+    if ((ctx->ones >= 10275) || (ctx->ones <= 9725)) {
+        rng_test |= FIPS_RNG_MONOBIT;
+    }
+
+    /* Poker calcs */
+    for (i = 0, j = 0; i < 16; i++) {
+        j += ctx->poker[i] * ctx->poker[i];
+    }
+
+    /* 16/5000*1563176-5000 = 2.1632  */
+    /* 16/5000*1576928-5000 = 46.1696 */
+    if ((j > 1576928) || (j < 1563176)) {
+        rng_test |= FIPS_RNG_POKER;
+    }
+
+    if ((ctx->runs[0] < 2315) || (ctx->runs[0] > 2685) ||
+        (ctx->runs[1] < 1114) || (ctx->runs[1] > 1386) ||
+        (ctx->runs[2] < 527) || (ctx->runs[2] > 723) ||
+        (ctx->runs[3] < 240) || (ctx->runs[3] > 384) ||
+        (ctx->runs[4] < 103) || (ctx->runs[4] > 209) ||
+        (ctx->runs[5] < 103) || (ctx->runs[5] > 209) ||
+        (ctx->runs[6] < 2315) || (ctx->runs[6] > 2685) ||
+        (ctx->runs[7] < 1114) || (ctx->runs[7] > 1386) ||
+        (ctx->runs[8] < 527) || (ctx->runs[8] > 723) ||
+        (ctx->runs[9] < 240) || (ctx->runs[9] > 384) ||
+        (ctx->runs[10] < 103) || (ctx->runs[10] > 209) ||
+        (ctx->runs[11] < 103) || (ctx->runs[11] > 209)) {
+        rng_test |= FIPS_RNG_RUNS;
+    }
+
+    /* finally, clear out FIPS variables for start of next run */
+    memset(ctx->poker, 0, sizeof(ctx->poker));
+    memset(ctx->runs, 0, sizeof(ctx->runs));
+    ctx->ones = 0;
+    ctx->rlength = -1;
+    ctx->current_bit = 0;
+
+    return rng_test;
+}
+
+
+static void do_rng_fips_test_loop(void)
+{
+    int j;
+    int fips_result;
+    struct timeval start, stop, statdump;
+    unsigned long int statruns, runs;
+    uint32_t overflow = 0U;
+    uint32_t value = 0U;
+
+    struct rng_stat *stat;
+
+    runs = statruns = 0;
+    TS(statdump);
+
+    for (;;) {
+        TS(start);
+
+	random_read(rng_buffer, sizeof(rng_buffer));
+
+        /* Update timer stats (source_blockfill) */
+        TS(stop);
+        stat = (struct rng_stat *)&rng_stats.source_blockfill;
+        overflow = stat->num_samples;
+        value = TS_DIFF(start, stop);
+
+        if ((stat->min == 0) || (value < stat->min)) {
+            stat->min = value;
+        }
+
+        if (value > stat->max) {
+            stat->max = value;
+        }
+
+        if (++stat->num_samples > overflow) {
+            stat->sum += value;
+        }
+        else {
+            stat->sum = value;
+            stat->num_samples = 1;
+        }
+
+        /* Run FIPS rng tests */
+        TS(start);
+        fips_result = fips_run_rng_test(&fipsctx, &rng_buffer);
```

Comment:
Use ```int fips_result = ...``` here instead of declaring it at the top of the function (you only use this
variable in the for loop).
Don't worry about optimization. Your compiler will do that for you ;-)

BTW:  It is generally a good idea to run 

```
./dist/tools/cppcheck/check.sh master
```

from time to time.